cmd/install/assets/hypershift_operator.go

```diff
@@ -477,10 +490,28 @@ func (o HyperShiftServiceMonitor) Build() *prometheusoperatorv1.ServiceMonitor {
 				{
 					Interval: "30s",
 					Port:     "metrics",
+					RelabelConfigs: []*prometheusoperatorv1.RelabelConfig{
+						{
+							SourceLabels: []string{"__meta_kubernetes_pod_label_hypershift_openshift_io_control_plane_component"},
+							TargetLabel:  "hypershift_component",
+						},
+					},
 				},
 			},
 		},
 	}
+
+	// We identify individual clusters from testruns by namespace. The namespace for the operator is not randomized though,
+	// so we have to add an additionl label that is guranteed to be unique, otherwise we might get conflicting metrics if
+	// we ever have two periodics running at the same time and exporting metrics.
+	if prowJobID := os.Getenv("PROW_JOB_ID"); prowJobID != "" {
+		monitor.Spec.Endpoints[0].RelabelConfigs = append(monitor.Spec.Endpoints[0].RelabelConfigs, &prometheusoperatorv1.RelabelConfig{
+			TargetLabel: "prowjob",
+			Replacement: prowJobID,
+		})
+	}
```

Comment:
I think sticking to the prowjob id is better, because it is clear in meaning. We have too many clusters in an e2e: The hosting one and the ones we create. Additionally, this allows us to find the prowjob which might be useful in case of outliers.

Moving it to the remote_write config is a great idea, I've done that.